### DISCUSSION TITLE
make carla for windows11

### DISCUSSION THREAD

**Discussion by 214122** [carla-simulator/carla#7731] [Q&A]
curl: (35) Recv failure: Connection was reset
tar: Error opening archive: Failed to open 'OSM2ODR.zip'
找不到 E:\carla\carla\Build\OSM2ODR.zip
系统找不到指定的文件。
CMake Error: The source directory "E:/carla/carla/Build/osm2odr-source"" does not exist.
Specify --help for usage, or press the help button on the CMake GUI.
Error: could not load cache
系统找不到指定的文件。
系统找不到指定的文件。
-[BuildOSM2ODR]: OSM2ODR has been successfully installed in "E:\carla\carla\PythonAPI\carla\dependencies\"
-[BuildPythonAPI]: [Batch params]: --py3
Building Python API for Python 3.
compiling:
  - source/libcarla/libcarla.cpp
running bdist_egg
running egg_info
writing source\carla.egg-info\PKG-INFO
writing dependency_links to source\carla.egg-info\dependency_links.txt
writing top-level names to source\carla.egg-info\top_level.txt
reading manifest file 'source\carla.egg-info\SOURCES.txt'
writing manifest file 'source\carla.egg-info\SOURCES.txt'
installing library code to build\bdist.win-amd64\egg
C:\python3.8\lib\site-packages\setuptools\_distutils\cmd.py:66: SetuptoolsDeprecationWarning: setup.py install is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` directly.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
        ********************************************************************************

!!
  self.initialize_options()
running install_lib
running build_py
copying source\carla\command.py -> build\lib.win-amd64-cpython-38\carla
copying source\carla\__init__.py -> build\lib.win-amd64-cpython-38\carla
running build_ext
building 'carla.libcarla' extension
E:\visual\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -Idependencies/include -IC:\python3.8\include -IC:\python3.8\Include -IE:\visual\VC\Tools\MSVC\14.29.30133\ATLMFC\include -IE:\visual\VC\Tools\MSVC\14.29.30133\include "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" "-IE:\Windows Kits\10\include\10.0.22621.0\ucrt" "-IE:\Windows Kits\10\include\10.0.22621.0\shared" "-IE:\Windows Kits\10\include\10.0.22621.0\um" "-IE:\Windows Kits\10\include\10.0.22621.0\winrt" "-IE:\Windows Kits\10\include\10.0.22621.0\cppwinrt" -IE:\visual\VC\Tools\MSVC\14.29.30133\ATLMFC\include -IE:\visual\VC\Tools\MSVC\14.29.30133\include "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" "-IE:\Windows Kits\10\include\10.0.22621.0\ucrt" "-IE:\Windows Kits\10\include\10.0.22621.0\shared" "-IE:\Windows Kits\10\include\10.0.22621.0\um" "-IE:\Windows Kits\10\include\10.0.22621.0\winrt" "-IE:\Windows Kits\10\include\10.0.22621.0\cppwinrt" /EHsc /Tpsource/libcarla/libcarla.cpp /Fobuild\temp.win-amd64-cpython-38\Release\source/libcarla/libcarla.obj /experimental:external /external:W0 /external:I dependencies/include/system /DBOOST_ALL_NO_LIB /DBOOST_PYTHON_STATIC_LIB /DBOOST_ERROR_CODE_HEADER_ONLY /D_WIN32_WINNT=0x0600 /DHAVE_SNPRINTF /DLIBCARLA_WITH_PYTHON_SUPPORT -DLIBCARLA_IMAGE_WITH_PNG_SUPPORT=true /MD
libcarla.cpp
dependencies/include\carla/road/element/RoadInfoMarkRecord.h(1): warning C4819: 该文件包含不能在当前代码页(936)中表示的 字符。请将该文件保存为 Unicode 格式以防止数据丢失
E:\carla\carla\PythonAPI\carla\source\libcarla\OSM2ODR.cpp(7): fatal error C1083: 无法打开包括文件: “OSM2ODR.h”: No such file or directory
error: command 'E:\\visual\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2

-[BuildPythonAPI]: Carla lib for python has been successfully installed in "E:\carla\carla\PythonAPI\carla\dist"!

**Comment by arscode** [comment]
better use pre-build package. I'm using it in windows 11, Carla 0.9.15

**Reply by 214122** [reply]
My circumstances are the same as yours, but I still have this problem

**Reply by sun12yyds** [reply]
are you ok

**Reply by sun12yyds** [reply]
..........
system repere

**Comment by yuhayang** [comment]
Have you solved the problem? I'm facing the same issue.

**Comment by Jinviz** [comment]
I found a solution. Open that header file as notepad, save it as UTF-8 (BOM) and rebuild it!

Was it helpful?
